src/MacroExpander.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * This file contains the “gullet” where macros are expanded
+ * until only non-macro tokens remain.
+ */
+
+var Lexer = require("./Lexer");
+
+function MacroExpander(input, macros) {
+    this.lexer = new Lexer(input);
+    this.macros = macros || {};
+    this.stack = []; // contains tokens in REVERSE order
+    this.discardedWhiteSpace = [];
+}
+
+MacroExpander.prototype.nextToken = function() {
```

Comment:
It be good to have a comment here about how this function expands macros until it can't and then returns the next atom.  I was thinking it might be clearer to have a separate function that just does the expansion art, thoughts?